pkg/localstore/localstore.go

```diff
@@ -292,9 +331,28 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
 		return nil, err
 	}
 
+	// instantiate sharky instance
+	var sharkyBase fs.FS
+	if path == "" {
+		sharkyBase = &memFS{Fs: afero.NewMemMapFs()}
+	} else {
+		sharkyBasePath := filepath.Join(path, "sharky")
+		if _, err := os.Stat(sharkyBasePath); os.IsNotExist(err) {
+			err := os.Mkdir(sharkyBasePath, 0775)
+			if err != nil {
+				return nil, err
+			}
+		}
+		sharkyBase = &dirFS{basedir: sharkyBasePath}
+	}
+	db.sharky, err = sharky.New(sharkyBase, sharkyNoOfShards, sharkyPerShardLimit)
+	if err != nil {
+		return nil, err
+	}
+
 	// Index storing actual chunk address, data and bin id.
 	headerSize := 16 + postage.StampSize
-	db.retrievalDataIndex, err = db.shed.NewIndex("Address->StoreTimestamp|BinID|BatchID|BatchIndex|Sig|Data", shed.IndexFuncs{
+	db.retrievalDataIndex, err = db.shed.NewIndex("Address->StoreTimestamp|BinID|BatchID|BatchIndex|Sig|Location", shed.IndexFuncs{
```

Comment:
Just to mention that this (and gcIndex) change requires a migration. Probably for the followup PR.

Reply:
thanks @janos, there's already a migration PR in progress. But why is a migration to gcIndex required? I might have overlooked something

Reply:
gcIndex is also changed  to include the location in value. The location should be appended to every value in that index.